utils/hct/hctbuild.cmd

```diff
@@ -184,13 +179,20 @@ if "%1"=="-dxc-cmake-ends-include" (
   shift /1
 )
 
-rem If only VS 2017 is available, pick that by default.
-if "%BUILD_VS_VER%"=="2015" (
-  reg query HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\DevDiv\vs\Servicing\14.0\devenv /v Install /reg:32 1>nul 2>nul
-  if errorlevel 1 (
-    echo Visual Studio 2015 not available, setting up build for Visual Studio 2017.
-    set BUILD_GENERATOR=Visual Studio 15 2017
-    set BUILD_VS_VER=2017
+rem If only VS 2019 is available, pick that by default.
+if "%BUILD_VS_VER%"=="2017" (
+  if not exist "%ProgramFiles(x86)%\Microsoft Visual Studio\2017" (
```

Comment:
I wonder if `%ProgramFiles(x86%` is not set on AppVeyor somehow.  It always seems to precede `\Microsoft` which looks like what it's complaining about:
```
call utils\hct\hctbuild -%PLATFORM% -%CONFIGURATION% -vs2017 -spirvtest
\Microsoft was unexpected at this time.
```

Reply:
I don't think that is the case. I have a local repro and it is caused by commit 01127626 (Codereview Feedback). It might be some batch file quirkiness.

Reply:
Quirkiness = it works when I comment out the four 'echo' commands on lines 185-6 and 193-4.